language/move-prover/src/spec_translator.rs

```diff
@@ -972,6 +977,51 @@ impl<'env> SpecTranslator<'env> {
         self.global_env()
             .set_condition_info(loc.clone(), tag, ConditionInfo::for_message(message));
     }
+
+    /// Assumes preconditions for function. This is used for the top-level verification
+    /// entry point of a function.
+    pub fn assume_preconditions(&self) {
```

Comment:
There is a new type ConditionDistribution which handles this. It might have been introduced after you have written this. This current treatment is not consistent with how we deal with conditions.